src/sentry/dynamic_sampling/tasks/custom_rule_notifications.py

```diff
@@ -0,0 +1,166 @@
+
+    for rule in TimedIterator(context, custom_active_rules, name="custom_rule_notifications"):
+        if rule.organization_id != cur_org_id:
+            cur_org_id = rule.organization_id
+            log_state.num_orgs += 1
+        log_state.num_rows_total += 1
+
+        num_samples = get_num_samples(rule)
+
+        if num_samples >= MIN_SAMPLES_FOR_NOTIFICATION:
+            send_notification(rule, num_samples)
+            rule.notification_sent = True
+            rule.save()
+
+
+def get_num_samples(rule: CustomDynamicSamplingRule) -> int:
+    """
+    Returns the number of samples accumulated for the given rule.
+    """
+    projects = rule.projects.all()
+
+    if not projects:
+        # org rule get all projects for org
+        projects = rule.organization.project_set.filter(status=ObjectStatus.ACTIVE)
+
+    params: Dict[str, Any] = {
+        "start": rule.start_date,
+        "end": rule.end_date,
+        "project_id": [p.id for p in projects],
+        "project_objects": projects,
+        "organization_id": rule.organization.id,
+    }
+
+    result = discover.query(
+        selected_columns=["count()"],
+        params=params,
+        query=rule.query,
+        referrer="dynamic_sampling.tasks.custom_rule_notifications",
+    )
+
+    return result["data"][0]["count"]
+
+
+def send_notification(rule: CustomDynamicSamplingRule, num_samples: int) -> None:
+    """
+    Notifies the rule creator that samples have been gathered.
+    """
+    subject_template = "We've collected {num_samples} samples for the query: {query} you made"
+
+    user_id = rule.created_by_id
+    if not user_id:
+        return
+
+    projects = rule.projects.all()
+    project_ids = [p.id for p in projects]
+
+    creator = User.objects.get_from_cache(id=user_id)
+    params = {
+        "query": rule.query,
+        "num_samples": num_samples,
+        "start_date": rule.start_date.strftime("%Y-%m-%d %H:%M:%S"),
+        "end_date": rule.end_date.strftime("%Y-%m-%d %H:%M:%S"),
+        "name": creator.name,
+        "email": creator.email,
+        "user_name": creator.username,
+        "display_name": creator.get_display_name(),
+        "discover_link": create_discover_link(rule, project_ids),
+    }
+
+    subject = subject_template.format(**params)
+
+    msg = MessageBuilder(
+        subject=subject,
+        template="sentry/emails/dyn-sampling-custom-rule-fulfilled.txt",
+        html_template="sentry/emails/dyn-sampling-custom-rule-fulfilled.html",
+        context=params,
+    )
+
+    msg.send_async([creator.email])
+
+
+def create_discover_link(rule: CustomDynamicSamplingRule, projects: List[int]) -> str:
+    """
+    Creates a discover link for the given rule.
+    It will point to a discover query using the same query as the rule
+    and the same time range as the rule.
+    """
+    url = absolute_uri(
+        reverse("sentry-api-0-discover-homepage-query", args=[rule.organization.slug])
```

Comment:
Are you looking to link to the API endpoint or the UI view? Right now you're generating a link to the API endpoint.

Reply:
You are right ! Thanks for pointing it out.
I am looking for the UI endpoint

Reply:
I thought so. There isn't an existing route name for the discover results view. You can either add a new route to `sentry/web/urls.py` so that `reverse()` works, or use a string for the path.

```python
discover_url = rule.organization.absolute_url(f"/organizations/{rule.organization.slug}/discover/results/", query=query_string)
```

Using `organization.absolute_url` helps ensure that URLs are correct in both saas (customer-domains on) and in single-tenant/self-hosted (customer-domains off).